js/components/thumbnail.js

```diff
@@ -4,15 +4,126 @@
+      var newThumbnailHeight = ReactDOM.findDOMNode(this.refs.thumbnail).clientHeight;
+      if (newThumbnailWidth !== this.thumbnailWidth || newThumbnailHeight !== this.thumbnailHeight) {
+        this.thumbnailWidth = newThumbnailWidth;
+        this.thumbnailHeight = newThumbnailHeight;
+        var yaw = this.props.vrViewingDirection.yaw;
+        var pitch = this.props.vrViewingDirection.pitch;
+        this.setCurrentViewVr(yaw, pitch);
+      }
+    }
+  },
+
+  setThumbnailSizes: function() {
+    var thumbnailWidth = ReactDOM.findDOMNode(this.refs.thumbnail).clientWidth;
+    var thumbnailHeight = ReactDOM.findDOMNode(this.refs.thumbnail).clientHeight;
+    if (thumbnailWidth) {
+      this.thumbnailWidth = thumbnailWidth;
+    }
+    if (thumbnailHeight) {
+      this.thumbnailHeight = thumbnailHeight;
+    }
+  },
+
+  setImageSizes: function() {
+    var thumbnail = Utils.findThumbnail(this.props.thumbnails, this.props.hoverTime, this.props.duration, this.props.videoVr);
+    if (thumbnail !== null && typeof thumbnail === 'object') {
+      var imageWidth = thumbnail.imageWidth;
+      var imageHeight = thumbnail.imageHeight;
+      if (imageWidth && imageHeight) {
+        if (imageWidth > 380) {
+          imageWidth = 380;
+          imageHeight = thumbnail.imageHeight * 380 / thumbnail.imageWidth;
+        }
+        this.imageWidth = imageWidth;
+        this.imageHeight = imageHeight;
+      }
+    }
+  },
+
+  /**x
+   * @description set positions for a thumbnail image when a video is vr
+   * @param {Number} yaw - rotation around the vertical axis in degrees (returns after changing direction)
+   * @param {Number} pitch - rotation around the horizontal axis in degrees (returns after changing direction)
+   * @private
+   */
+  setCurrentViewVr: function(yaw, pitch) {
+    yaw = Utils.ensureNumber(yaw, 0);
+    pitch = Utils.ensureNumber(pitch, 0);
+    var imageWidth = this.imageWidth;
+    var imageHeight = this.imageHeight;
+    var thumbnailWidth = this.thumbnailWidth;
+    var thumbnailHeight = this.thumbnailHeight;
+    yaw = this.getCurrentYawVr(yaw);
+    pitch = pitch >= 360 ? 0 : pitch;
+
+    var positionY = -(((imageHeight - thumbnailHeight) / 2) - pitch);
+    var bottomCoordinate = -(imageHeight - thumbnailHeight);
+    if (positionY > 0) {
+      positionY = 0;
+    } else if (positionY < bottomCoordinate) {
+      positionY = bottomCoordinate;
+    }
+    var positionX = -(imageWidth - thumbnailWidth / 2 - imageWidth * yaw / 360);
+    this.positionY = positionY;
+    this.positionX = positionX;
+  },
+
+  /**
+   * @description return current coefficient of the yaw if yaw > 360 or yaw < -360 degrees
+   * @param {Number} yaw - angle in degrees
+   * @private
+   * @returns {number} coefficient showing how many times to take 360 degrees
+   */
+  getCurrentYawVr: function(yaw) {
+    var k = yaw <= -360 ? -1 : 1;
+    var ratio = k * yaw / 360;
+    ratio = ~~ratio;
```

Comment:
How come we want to floor the ratio?

Reply:
bitmovin can  return a value like 764, but we need only value from 0 to 360. So, in example 764 is equal 44 : 764 - 2*360, in this case 2 is ratio. And ratio is the largest integer divisor.